middleware/proxy/reverseproxy.go

```diff
@@ -153,8 +155,37 @@ func (p *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request, extr
 
 	copyHeader(rw.Header(), res.Header)
 
-	rw.WriteHeader(res.StatusCode)
-	p.copyResponse(rw, res.Body)
+	if res.StatusCode == HTTPSwitchingProtocols && outreq.Header.Get("Upgrade") == "websocket" {
```

Comment:
Got it, I'm convinced too.

@acmacalister Would you make the changes in your PR if you have a chance, today? Then I can just merge in one PR which is (probably) correct and commit my tests for it later today.